### PR TITLE
[processor/cumulativetodelta] Update stability level to beta

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
The cumulativetodelta processor's config was[ previously updated](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8952) to a stable state.  I'd like to bring this component's stability level in line with that change.  I know this processor is also getting production exposure, and I'd like to consider it "Stable" soon, but I don't want to jump from "In Developement" to "Stable".

This processor will likely be replaced by the transform processor, but that is a long way off.  In the meantime, I'd like customers to feel better about using this processor.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

**Comment by djaglowski** [comment]
#10005 should resolve the CI failures

**Comment by bogdandrutu** [comment]
I would not put "stable" if we know we want to replace.

**Comment by TylerHelmuth** [comment]
> I would not put "stable" if we know we want to replace.

Thats fair, but I think `Beta` is still appropriate at this time.

**Comment by djaglowski** [comment]
#10005 is merged. Please rebase

**Review by codeboten — APPROVED** [review]
Minor change, as suggested in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10006

File: processor/cumulativetodeltaprocessor/README.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 # Cumulative to Delta Processor
 
-**Status: under development; Not recommended for production usage.**
-
-Supported pipeline types: metrics
+| Status                   |           |
+| ------------------------ | --------- |
+| Stability                | [beta]    |
+| Supported pipeline types | metrics   |
+| Distribution             | [contrib] |
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply processor/cumulativetodeltaprocessor/README.md]: Changed

File: processor/cumulativetodeltaprocessor/README.md [hunk 2]
```diff
@@ -72,3 +74,6 @@ processors:
         # If include/exclude are not specified
         # convert all cumulative sum metrics to delta
 ```
+
+[beta]: https://github.com/open-telemetry/opentelemetry-collector-contrib#beta
```
Comment (pmm-sumo) [inline processor/cumulativetodeltaprocessor/README.md:78]: This link was wrong @TylerHelmuth, I think it should have been:

```
https://github.com/open-telemetry/opentelemetry-collector#beta
```

Comment (pmm-sumo) [reply processor/cumulativetodeltaprocessor/README.md:78]: Sure thing! Opened an issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10162

**Pull request merged**